### DISCUSSION TITLE
Integrate DocSearch x Algolia in a DocFx site

### DISCUSSION THREAD

**Discussion by cdervout** [dotnet/docfx#9668] [Q&A] [answered]
Hello, 

I'd like to integrate [Algolia ](https://docsearch.algolia.com/docs/DocSearch-v3) search into a site generated by Docfx using the Modern template. I am asked to :

1. Copy this snippet at the end of the HTML `<head>` tag
`<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@docsearch/css@3"/></pre></li>`

How update the  `<head>` tag generation ?

2. Copy this snippet at the end of the HTML `<body>` tag

```
<script src="https://cdn.jsdelivr.net/npm/@docsearch/js@3"></script>
<script type="text/javascript">
docsearch({
appId: xxxxxxx,
apiKey: xxxxxxx,
indexName: xxxxxxxxx,
insights: true, // Optional, automatically send insights when user interacts with search results
container: '### REPLACE ME WITH A CONTAINER (e.g. div) ###'
debug: false // Set debug to true if you want to inspect the modal
});
</script>
```

How update the  `<body>` tag generation ?
How do you determine the container div to be defined in this setting?

**Comment by filzrev** [comment, marked as answer]
It's need to create custom template for DocSearch.

docfx `modern` template don't support partial template update.
It need to copy **entire template layout file** and need to modify contents. 

**Required steps**

1. Create custom template that clone modern template's layout file.  
    https://github.com/filzrev/docfx.samples.docsearch/commit/3ef095eb922de544ea64405e68d7a91517aad4cd
2. Add required DocSearch settings.
    https://github.com/filzrev/docfx.samples.docsearch/commit/70d204c2d587fa4a2c1557e79af2d78a488ef2da

**Sample DocSearch enabled site (Using `Demo API key`)**
https://filzrev.github.io/docfx.samples.docsearch

**Comment by cdervout** [comment]
That's what I was afraid of. Thanks for the answer. I've got everything I need to move forward.
